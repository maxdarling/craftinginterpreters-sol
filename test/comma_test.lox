// Test comma operator (evaluates left to right, returns rightmost value)
print "=== Comma Operator Tests ===";

// Basic comma operations
print (1, 2); // 2
print (1, 2, 3); // 3
print ("hello", "world"); // "world"
print (true, false); // false

// With side effects
var x = 0;
print (x = 1, x = 2, x = 3); // 3
print x; // 3

// Mixed with other operators
print (1 + 2, 3 * 4); // 12
print (5 > 3, 2 < 1); // false
print (true and false, true or false); // true

// In variable assignment
var result = (10, 20, 30);
print result; // 30

// With function calls
fun sideEffect(value) {
  print "Side effect: " + value;
  return value;
}

print (sideEffect("first"), sideEffect("second")); // prints side effects, returns "second"

// Note: Comma in for loop variable declarations not supported
// Note: Multiple variable declarations in one statement not supported

// Comma with assignment (individual variables)
var a = 0;
var b = 0;
print (a = 1, b = 2); // 2
print a; // 1
print b; // 2

print "Comma tests completed";
