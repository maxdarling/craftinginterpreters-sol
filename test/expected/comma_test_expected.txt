=== Comma Operator Tests ===
2
3
world
false
3
3
12
false
true
30
Side effect: first
Side effect: second
second
2
1
2
Comma tests completed
